pilot/pkg/networking/core/v1alpha3/sidecar_simulation_test.go

```diff
@@ -27,6 +27,476 @@ import (
+					Port:     80,
+					Protocol: simulation.HTTP,
+					CallMode: simulation.CallModeInbound,
+				},
+				Result: simulation.Result{
+					VirtualHostMatched: "inbound|http|80",
+					ClusterMatched:     "inbound|80|http|foo.bar",
+				},
+			},
+			{
+				Name: "auto port inbound",
+				Call: simulation.Call{
+					Port:     81,
+					Protocol: simulation.HTTP,
+					CallMode: simulation.CallModeInbound,
+				},
+				Result: simulation.Result{
+					VirtualHostMatched: "inbound|http|81",
+					ClusterMatched:     "inbound|81|auto|foo.bar",
+				},
+			},
+			{
+				Name: "auto port2 inbound",
+				Call: simulation.Call{
+					Port:     81,
+					Protocol: simulation.HTTP2,
+					CallMode: simulation.CallModeInbound,
+				},
+				Result: simulation.Result{
+					VirtualHostMatched: "inbound|http|81",
+					ClusterMatched:     "inbound|81|auto|foo.bar",
+				},
+			},
+			{
+				Name: "auto tcp inbound",
+				Call: simulation.Call{
+					Port:     81,
+					Protocol: simulation.TCP,
+					CallMode: simulation.CallModeInbound,
+				},
+				Result: simulation.Result{
+					ListenerMatched:    "virtualInbound",
+					FilterChainMatched: "0.0.0.0_81",
+					ClusterMatched:     "inbound|81|auto|foo.bar",
+					ListenerTLS:        simulation.TlsModePassthrough,
+					StrictMatch:        true,
+				},
+			},
+			{
+				Name: "passthrough http",
+				Call: simulation.Call{
+					Address:  "1.2.3.4",
+					Port:     82,
+					Protocol: simulation.HTTP,
+					CallMode: simulation.CallModeInbound,
+				},
+				Result: simulation.Result{
+					ClusterMatched: "InboundPassthroughClusterIpv4",
+				},
+			},
+			{
+				Name: "passthrough tcp",
+				Call: simulation.Call{
+					Address:  "1.2.3.4",
+					Port:     82,
+					Protocol: simulation.TCP,
+					CallMode: simulation.CallModeInbound,
+				},
+				Result: simulation.Result{
+					ClusterMatched: "InboundPassthroughClusterIpv4",
+				},
+			},
+			{
+				Name: "passthrough tls",
+				Call: simulation.Call{
+					Address:  "1.2.3.4",
+					Port:     82,
+					Protocol: simulation.TLS,
+					CallMode: simulation.CallModeInbound,
+				},
+				Result: simulation.Result{
+					ClusterMatched: "InboundPassthroughClusterIpv4",
+				},
+			},
+		},
+	})
+
+	runSimulationTest(t, nil, xds.FakeOptions{}, simulationTest{
+		name:           "permissive",
+		config:         svc + mtlsMode("PERMISSIVE"),
+		skipValidation: true, // TODO(https://github.com/istio/istio/issues/26079)
+		calls: []simulation.Expect{
+			{
+				Name: "http port",
+				Call: simulation.Call{
+					Port:     80,
+					Protocol: simulation.HTTP,
+					CallMode: simulation.CallModeInbound,
+				},
+				Result: simulation.Result{
+					VirtualHostMatched: "inbound|http|80",
+					ClusterMatched:     "inbound|80|http|foo.bar",
+					ListenerTLS:        simulation.TlsModePassthrough,
+				},
+			},
+			{
+				Name: "http port tls",
+				Call: simulation.Call{
+					Port:     80,
+					Protocol: simulation.HTTPS,
+					Alpn:     "http/1.1",
```

Comment:
I don't get the Protocol here. For incoming traffic it's the combination 
* transport-socket: tls or raw_buffer
* alpn: http/1.1, http/1.0, h/2, istio, "istio", "istio-http/1.0", "istio-http/1.1", "istio-h2"

This expect input parameter doesn't fit my mental model

Reply:
I am considering simuation.HTTPS a TLS transport protocol and it looks good so far :)

Reply:
Yes I think its a bit weird in this case - it made more sense when I originally wrote this code which was for gateways. How it works in the filterchain match logic:
```
func(fc *listener.FilterChainMatch) bool {
		if !hasTlsInspector {
			// Without tls inspector, transport protocol will always be raw buffer
			return fc.GetTransportProtocol() == xdsfilters.RawBufferTransportProtocol
		}
		switch fc.GetTransportProtocol() {
		case xdsfilters.TLSTransportProtocol:
			return input.Protocol == HTTPS || input.Protocol == TLS
		case xdsfilters.RawBufferTransportProtocol:
			return input.Protocol != HTTPS && input.Protocol != TLS
		}
		return false
	})
```

let me see if I can clean this up

Reply:
closing in favor of https://github.com/istio/istio/pull/28083 which cleans it up